### PR TITLE
GEOPY-493

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-493 - Add Run test for ZTEM inversion**

### CODE REVIEW

File: geoapps/inversion/driver.py [hunk 1]
```diff
@@ -168,21 +173,16 @@ def initialize(self):
...
-        # Solve forward problem, and attach dpred to inverse problem or
+        # If forward only option enabled, stop here
         if self.params.forward_only:
-            print("Running forward simulation ...")
-        else:
-            print("Pre-computing sensitivities ...")
+            return
 
-        if self.warmstart or self.params.forward_only:
+        if self.warmstart:
```
Comment (benk-mira) [inline geoapps/inversion/driver.py:180]: Bulking up the logging statements - Nice!

File: geoapps/inversion/natural_sources/tipper/constants.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    "receivers_offset_z": 0.0,
+    "gps_receivers_offset": None,
+    "resolution": None,
+    "max_chunk_size": 128,
+    "chunk_by_rows": True,
+    "mesh": None,
+    "u_cell_size": 25.0,
+    "v_cell_size": 25.0,
+    "w_cell_size": 25.0,
+    "octree_levels_topo": [16, 8, 4, 2],
```
Comment (benk-mira) [inline geoapps/inversion/natural_sources/tipper/constants.py:145]: [0, 0, 4, 4]?

File: tests/run_tests/driver_dc_test.py [hunk 3]
```diff
@@ -48,17 +48,17 @@ def test_dc_run(
         flatten=False,
     )
 
-    tx_obj = geoh5.get_entity("survey (currents)")[0]
-    tx_obj.cells = tx_obj.cells.astype("uint32")
+    # tx_obj = geoh5.get_entity("survey (currents)")[0]
```
Comment (benk-mira) [inline tests/run_tests/driver_dc_test.py:51]: cleanup

File: tests/run_tests/driver_grav_test.py [hunk 4]
```diff
@@ -37,7 +37,7 @@ def test_gravity_run(
 
     np.random.seed(0)
     # Run the forward
-    geoh5 = setup_inversion_workspace(
+    geoh5, mesh, model, survey, topography = setup_inversion_workspace(
```
Comment (benk-mira) [inline tests/run_tests/driver_grav_test.py:40]: smart!

**Review by benk-mira — APPROVED** [review]
Mostly cleanups here - Lookin great!

File: geoapps/inversion/natural_sources/tipper/constants.py (hunk 2 above)
Comment (domfournier) [reply geoapps/inversion/natural_sources/tipper/constants.py:145]: Definite no. We want to refine tight on the obs, but care less about topo. Topography is the most expensive distretization.

Comment (benk-mira) [reply geoapps/inversion/natural_sources/tipper/constants.py:145]: This was a suggestion for topo.  instead of 16, 8, 4, 2.

Comment (domfournier) [reply geoapps/inversion/natural_sources/tipper/constants.py:145]: Oh yep

Comment (domfournier) [reply geoapps/inversion/natural_sources/tipper/constants.py:145]: Done

File: tests/run_tests/driver_dc_test.py (hunk 3 above)
Comment (domfournier) [reply tests/run_tests/driver_dc_test.py:51]: Done

File: geoapps/inversion/components/factories/directives_factory.py [hunk 5]
```diff
@@ -97,7 +99,17 @@ def build(
                 active_cells=active_cells,
                 name="Model",
             )
-
+            # self.save_iteration_sensitivities_directive = SaveIterationGeoh5Factory(
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/factories/directives_factory.py:102]: Done

File: geoapps/inversion/components/factories/directives_factory.py [hunk 6]
```diff
@@ -39,6 +40,7 @@ def __init__(self, params):
         self.update_preconditioner_directive = None
         self.save_iteration_model_directive = None
         self.save_iteration_data_directive = None
+        # self.save_iteration_sensitivities_directive = None
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/factories/directives_factory.py:43]: Done

File: geoapps/inversion/components/factories/directives_factory.py [hunk 7]
```diff
@@ -23,6 +23,7 @@ class DirectivesFactory:
         "SaveIterationsGeoH5": [
             "save_iteration_model_directive",
             "save_iteration_data_directive",
+            # "save_iteration_sensitivities_directive",
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/factories/directives_factory.py:26]: Done

File: geoapps/inversion/components/data.py [hunk 8]
```diff
@@ -487,8 +490,12 @@ def simulation(
                 minimum_level=3,
                 padding_cells=padding_cells,
             )
+
             kwargs = {"components": 3} if self.vector else {}
-            map = maps.TileMap(mesh, active_cells, nested_mesh, **kwargs)
+            map = maps.TileMap(
+                mesh, active_cells, nested_mesh, enforce_active=True, **kwargs
+            )
+            # map._local_mesh = None
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/data.py:498]: Done

File: geoapps/inversion/components/data.py [hunk 9]
```diff
@@ -497,7 +504,7 @@ def simulation(
                 map=map,
                 tile_id=tile_id,
             )
-
+            # del nested_mesh
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/data.py:507]: Done

File: geoapps/inversion/components/factories/entity_factory.py [hunk 10]
```diff
@@ -41,27 +41,27 @@ def factory_type(self):
...
+        else:
+            return type(self.params.data_object)
+
     def build(self, inversion_data: InversionData):
         """Constructs geoh5py object for provided inversion type."""
 
         if self.factory_type in ["direct current", "induced polarization"]:
             return self._build_dcip(inversion_data)
+        # elif self.factory_type in ["tipper"]:
+        #     return self._build_tipper(inversion_data)
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/components/factories/entity_factory.py:64]: Done

**Pull request merged**